src/adr/0072-postgresql-flags-update-concurrency-control.md

```diff
@@ -0,0 +1,50 @@
+# 72.  Postgresql flags update concurrency control mechanism
+
+Date: 2023-12-19
+
+## Status
+
+Not-Implemented
+
+## Context
+
+When IMAP server support THREAD_SAFE_FLAGS_UPDATE capability, it means that the server can update flags concurrently.
```

Comment:
No. We do not get the choice.

To get a working IMAP server we need to support concurrency races upon flag updates and get a decent behaviour.